### ISSUE TITLE
Numpy and JAX behaviours differ for sums over scalars.

### ISSUE THREAD

**Issue by AdrienCorenflos** [google/jax#14625]
### Description

When passed a scalar, `jax.numpy` cannot sum over the first nor last axis.

```python
import jax.numpy as jnp
import numpy as np

arr = 0.  # An array that is scalar for some reason
np.sum(arr, 0)  # no pb
np.sum(arr, -1)  # no pb

jnp.sum(arr, 0)  # raises
jnp.sum(arr, -1)  # raises
```

This is also valid for the `jnp.prod` function.
I believe this should likely be a valid input to `jnp.sum` WDYT?

### What jax/jaxlib version are you using?

jax==0.4.3 jaxlib==0.4.3+cuda11.cudnn86

### Which accelerator(s) are you using?

CPU/GPU

**Comment by jakevdp** [comment]
Thanks for the report – I don't see this behavior documented anywhere in numpy, so I suspect it's an implementation detail rather than intended design. I think JAX's behavior makes more sense: in general if you have an array `a` and you sum over `axis=a.ndim`, you will get an error. I think this also makes sense when `ndim = 0`. What do you think?

**Comment by soraros** [comment]
@jakevdp Should we also raise for `cumsum` and friends

```python
jnp.cumsum(0, axis=2)
# Array([0], dtype=int32, weak_type=True)
```
The NumPy version behaviours slightly differently, and raises for `axis not in {0, -1}`. Both versions violates the documentation

> The result has the same size as `a`, and the same shape as `a` if `axis` is not None or `a` is a 1-d array.

Sometimes NumPy's semantics is just 🤦 .

**Comment by jakevdp** [comment]
That looks like a bug in the implementation of `jnp.cumsum`. Are you interested in putting together a fix?

**Comment by soraros** [comment]
> Are you interested in putting together a fix?
@jakevdp Sure.

**Comment by AdrienCorenflos** [comment]
I've checked other stats function. For instance `np.mean` raises on scalars when an axis is passed. It seems to confirm that this is more of an implementation side effect than a feature of NumPy.
